### PR TITLE
Finalizing state

### PR DESCRIPTION
**Pull request by fishkerez**

### CODE REVIEW

File: runtime_scan/pkg/scanner/job_managment.go [hunk 1]
```diff
@@ -49,17 +50,19 @@ func (s *Scanner) jobBatchManagement(scanDone chan struct{}) {
 	imagesCompletedToScan := &s.progress.ImagesCompletedToScan
 	s.Unlock()
 
-	// queue of jobs
+	// queue of scan data
```
Comment (akpsgit) [inline runtime_scan/pkg/scanner/job_managment.go:53]: each scan data is related to a job

File: runtime_scan/pkg/scanner/job_managment.go [hunk 2]
```diff
@@ -74,6 +77,7 @@ func (s *Scanner) jobBatchManagement(scanDone chan struct{}) {
 		fullScanDone <- true
 	}()
 
+	// send all data on data queue, for workers to pick it up.
```
Comment (akpsgit) [inline runtime_scan/pkg/scanner/job_managment.go:80]: send all scan data on scan data queue

File: runtime_scan/pkg/scanner/job_managment.go [hunk 3]
```diff
@@ -86,16 +90,19 @@ func (s *Scanner) jobBatchManagement(scanDone chan struct{}) {
...
 		log.WithFields(s.logFields).Info("Scan process was canceled")
 	case <-fullScanDone:
 		log.WithFields(s.logFields).Infof("All jobs has finished")
 		// Nonblocking notification of a finished scan
+		// notify back to server on scan done
 		nonBlockingNotification(scanDone)
 	}
 }
 
+// worker waits for data on the queue, run a scan job and waits for results from that scan job. he then notify that he is done to the Scanner
```
Comment (akpsgit) [inline runtime_scan/pkg/scanner/job_managment.go:105]: // worker waits for data on the queue, runs a scan job and waits for results from that scan job. Upon completion, done is notified to the caller

File: runtime_scan/pkg/scanner/scanner.go [hunk 4]
```diff
@@ -173,6 +173,7 @@ func (s *Scanner) shouldIgnorePod(pod *corev1.Pod) bool {
 	return false
 }
 
+// initScan prepare scan config for all images
```
Comment (akpsgit) [inline runtime_scan/pkg/scanner/scanner.go:176]: Calculate properties of scan targets

File: runtime_scan/pkg/scanner/job_managment.go [hunk 5]
```diff
@@ -86,16 +90,19 @@ func (s *Scanner) jobBatchManagement(scanDone chan struct{}) {
...
 	}
 
+	// wait for killSignal or fullScanDone
 	select {
 	case <-s.killSignal:
 		log.WithFields(s.logFields).Info("Scan process was canceled")
 	case <-fullScanDone:
 		log.WithFields(s.logFields).Infof("All jobs has finished")
 		// Nonblocking notification of a finished scan
+		// notify back to server on scan done
```
Comment (akpsgit) [inline runtime_scan/pkg/scanner/job_managment.go:100]: job management shouldn't be aware that there is a server.

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,5 @@
 **/*.so
+.DS_Store
```
Comment (akpsgit) [inline .gitignore:2]: Please also delete the .DS_store file from the repo

Comment (fishkerez) [reply .gitignore:2]: what do you mean?

File: runtime_scan/pkg/scanner/job_managment.go (hunk 1 above)
Comment (fishkerez) [reply runtime_scan/pkg/scanner/job_managment.go:53]: its a queue of scan data though.. scan data is not related to a specific job currently

File: runtime_scan/pkg/scanner/job_managment.go (hunk 5 above)
Comment (fishkerez) [reply runtime_scan/pkg/scanner/job_managment.go:100]: he is not, the comment meant to help the maintainers of the code to understand better

Comment (akpsgit) [reply runtime_scan/pkg/scanner/job_managment.go:100]: not a good thing to put comments in an infra code about specific implementation of users.

File: runtime_scan/pkg/scanner/job_managment.go (hunk 1 above)
Comment (akpsgit) [reply runtime_scan/pkg/scanner/job_managment.go:53]: why not related. each job created for each image and it's all managed in scanData (one per job) :
```
type scanData struct {
	imageHash                    string
	imageID                      string
	contexts                     []*imagePodContext // All the pods that contain this image hash
	scanUUID                     string
```

**Review by akpsgit — APPROVED** [review]

**Pull request merged**